### PR TITLE
Service will retry once when it encounters 401 on token verification.

### PR DESCRIPTION
**Pull request by johnwu96822**
Switch to go mod.

Fix race condition by not deleting tokens from cache.

Fix exp time parsing format for service reading the expiration time from token verification result.

- [ ] @johnny-lai 
- [x] @abdulchaudhrycoupa 
- [ ] @supriya

## Generated Reviewers

<details>
  <summary><em>:warning: Do not modify anything in this section! :warning:</em></summary>
  <em>The content in this section is managed automatically, and any manual changes to it will be ignored and overwritten the next time the code review status is refreshed! View <a href="https://cody.coupa.engineering/repos/coupa/sand-go/pull/19">this PR</a> on Cody to see the current code review status.</em>
</details>

### CODE REVIEW

**Review by abdulchaudhrycoupa — APPROVED** [review]

File: sand.go [hunk 1]
```diff
@@ -17,13 +17,23 @@ import (
...
+
+	DefaultTokenOption = TokenOption{
+		NumRetry:        3,
+		SkipCachedToken: false,
+	}
 )
 
+type TokenOption struct {
+	NumRetry        int
+	SkipCachedToken bool
```
Comment (johnny-lai) [inline sand.go:34]: Minor: `SkipCachedToken` appears to mean it should not cache the token, rather than avoid "cached tokens". Perhaps `SkipTokenCaching`?

**Review by johnny-lai — APPROVED** [review]
Minor naming suggestion. lgtm otherwise. pre-approving

**Review by supriya — APPROVED** [review]

File: sand.go (hunk 1 above)
Comment (johnwu96822) [reply sand.go:34]: I think to keep the name

**Comment by johnwu96822** [comment]
```
AMER-M-611:sand-go johnwu$ go test
Running Suite: Sand Suite
=========================
Random Seed: 1629255114
Will run 65 of 65 specs

•••ERRO[0000] Post "": unsupported protocol scheme ""      
•ERRO[0000] Service failed to verify the token: Post "": unsupported protocol scheme "" 
•WARN[0000] Sand token: retrying after 1 sec because of error: Post "": unsupported protocol scheme "" 
WARN[0001] Sand token: retrying after 2 sec because of error: Post "": unsupported protocol scheme "" 
ERRO[0003] Post "": unsupported protocol scheme ""      
••••••••••••••••••••ERRO[0003] Error response from the authentication service: 500 -  
••••2021/08/18 10:51:57 http: TLS handshake error from 127.0.0.1:64582: tls: client offered only unsupported versions: [304 303]
2021/08/18 10:51:57 http: TLS handshake error from 127.0.0.1:64583: tls: client offered only unsupported versions: [304 303]
•2021/08/18 10:51:57 http: TLS handshake error from 127.0.0.1:64587: tls: client offered only unsupported versions: [304 303]
••••••••••••WARN[0003] Sand request: retrying after 1 sec on 401    
WARN[0004] Sand request: retrying after 2 sec on 401    
••••WARN[0006] Sand request: retrying after 1 sec on 401    
•WARN[0007] Sand request: retrying after 1 sec on 401    
WARN[0008] Sand request: retrying after 1 sec on 401    
•WARN[0009] Sand request: retrying after 1 sec on 401    
••••••••••WARN[0010] Sand token: retrying after 1 sec because of error: oauth2: cannot fetch token: 404 Not Found
Response:  
WARN[0011] Sand token: retrying after 2 sec because of error: oauth2: cannot fetch token: 404 Not Found
Response:  
•••••••
Ran 65 of 65 Specs in 13.067 seconds
SUCCESS! -- 65 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
ok  	github.com/coupa/sand-go	13.367s
```

**Review by supriya — APPROVED** [review]

**Review by johnny-lai — APPROVED** [review]

**Comment by johnwu96822** [comment]
[lockit v4.9.3] merged via Lockit CLI on 2021-08-23 14:11:06.382525 +0800 CST m=+5.259019704

**Pull request merged**